### PR TITLE
Jormungandr: Bragi calls

### PR DESCRIPTION
**Pull request by patochectp**
JIRA: http://jira.canaltp.fr/browse/NAVP-584

### CODE REVIEW

**Comment by antoine-de** [comment]
there are conflicts in your branch

File: source/jormungandr/jormungandr/autocomplete/elastic_search.py [hunk 1]
```diff
@@ -204,3 +208,124 @@ def get(self, request, instance):
...
+        geocoding = obj.get('properties', {}).get('geocoding', {})
+
+        housenumber = geocoding.get('housenumber')
+        return {
+            "id": geocoding.get('id'),
+            "coord": {
+                "lon": lon,
+                "lat": lat,
+            },
+            "house_number": housenumber if ((housenumber is not None) & (housenumber != "")) else "0",
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/elastic_search.py]: I think you can rewrite this with:

`"house_number": geocoding.get('housenumber') or '0'`

File: source/jormungandr/jormungandr/autocomplete/elastic_search.py [hunk 2]
```diff
@@ -204,3 +208,124 @@ def get(self, request, instance):
...
+}
+
+
+class GeocodejsonFeature(fields.Raw):
+    def format(self, place):
+        type_ = place.get('properties', {}).get('geocoding', {}).get('type')
+
+        if type_ == 'city':
+            return marshal(place, geocode_admin)
+        elif (type_ == 'street') | (type_ == 'house'):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/elastic_search.py]: I think you want to use `||` (or better `or`), `|` is  a binary `or`.

you can also do `elif type_ in ('street, 'house):`

and there is no need for the `()`

File: source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    assert navitia_response[0].get('embedded_type') == "address"
+    assert navitia_response[0].get('id') == "addr:49.847586;3.282103"
+    assert navitia_response[0].get('name') == "Rue Jean Jaures"
+    address = navitia_response[0].get('address', {})
+    assert address.get('name') == "Rue Jean Jaures"
+    assert address.get('house_number') == "20"
+    assert address.get('coord', {}).get('lat') == 49.847586
+    assert address.get('coord', {}).get('lon') == 3.282103
+    assert len(address.get('administrative_regions')) == 5
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py:78]: I think you should also check one admin

File: source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                            3.282103,
+                            49.847586
+                        ],
+                        "type": "Point"
+                    },
+                    "properties": {
+                        "geocoding": {
+                            "city": "Saint-Quentin",
+                            "id": "addr:49.847586;3.282103",
+                            "label": "20 Rue Jean Jaures, 02100 Saint-Quentin",
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py]: minor: there should not be `20` in the label :wink:

File: source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+    assert navitia_response[0].get('embedded_type') == "address"
+    assert navitia_response[0].get('id') == "addr:49.847586;3.282103"
+    assert navitia_response[0].get('name') == "Rue Jean Jaures"
+    address = navitia_response[0].get('address', {})
+    assert address.get('name') == "Rue Jean Jaures"
+    assert address.get('house_number') == "0"
+    assert address.get('coord', {}).get('lat') == 49.847586
+    assert address.get('coord', {}).get('lon') == 3.282103
+    assert len(address.get('administrative_regions')) == 5
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py]: same here, check one admin

File: source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+    mock_requests = MockRequests({
+        'http://bob.com/autocomplete?q=rue bobette':
+        (mock_good_geocodejson_response(), 200)
+    })
+
+    # we mock the http call to return the hard coded mock_response
+    with mock.patch('requests.get', mock_requests.get):
+        response = bragi.get({'q': 'rue bobette'}, instance=None)
+        # TODO assert on response
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py]: don't forget those TODOs too

File: source/jormungandr/jormungandr/tests/__init__.py [hunk 7]
```diff
@@ -0,0 +1 @@
+__author__ = 'antoine'
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/tests/__init__.py]: sorry my mistake, but you can remove this line and add the licence instead

**Comment by antoine-de** [comment]
Jenkins, test this please

File: source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py [hunk 8]
```diff
@@ -82,6 +82,12 @@ def bragi_house_reading_test():
     assert address.get('coord', {}).get('lat') == 49.847586
     assert address.get('coord', {}).get('lon') == 3.282103
     assert len(address.get('administrative_regions')) == 5
+    region_list = {region['level']: region['name'] for region in address.get('administrative_regions')}
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py:79]: :+1:

File: source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py [hunk 9]
```diff
@@ -131,6 +137,10 @@ def bragi_street_reading_test():
     assert address.get('coord', {}).get('lat') == 49.847586
     assert address.get('coord', {}).get('lon') == 3.282103
     assert len(address.get('administrative_regions')) == 5
+    response = next(region for region in address.get('administrative_regions'))
+    print(response)
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py]: you have a unused debug there

File: source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py (hunk 3 above)
Comment (pbougue) [inline source/jormungandr/jormungandr/autocomplete/tests/elastic_search_tests.py:78]: Done

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**